hooks/tk-multi-publish2/basic/publish_lookfiles.py

```diff
@@ -0,0 +1,426 @@
+        if index:
+            combo.setCurrentIndex(index)
+
+    def getData(self):
+        items = []
+        current_index = self._combo.currentIndex()
+        to_publish = self._combo.itemData(current_index)
+        for index in range(self._combo.count()):
+            path = self._combo.itemData(index)
+            items.append(path)
+        return {
+            self.__name: {
+                "work_paths": items,
+                "to_publish": to_publish
+            }
+        }
+
+
+class LookFilePublishWidget(QtGui.QWidget):
+    def __init__(self, parent=None, description_widget=None):
+        super(LookFilePublishWidget, self).__init__(parent=parent)
+        self.__contents = {}
+        self.__displayed = []
+        layout = QtGui.QVBoxLayout()
+        self.setLayout(layout)
+        if description_widget:
+            layout.addWidget(description_widget)
+        row = QtGui.QHBoxLayout()
+        row.addWidget(QtGui.QLabel("<b>Node Name</b>"))
+        row.addWidget(QtGui.QLabel("<b>Version</b>"))
+        layout.addLayout(row)
+
+    def clear(self):
+        for widget in self.__displayed:
+            self.layout().removeWidget(widget)
+            widget.setVisible(False)
+        self.__displayed = []
+
+    def addItem(self, name, data):
+        if name in self.__contents:
+            widget = self.__contents[name]
+        else:
+            widget = LookFilePublishWidgetItem(name, data, parent=self)
+            self.__contents[name] = widget
+        self.layout().addWidget(widget)
+        self.__displayed.append(widget)
+        widget.setVisible(True)
+
+    def getData(self):
+        data = {}
+        for widget in self.__displayed:
+            data.update(widget.getData())
+        return {"node_settings": data}
+
+
+class KatanaLookFilePublishPlugin(HookBaseClass):
+    """
+    Plugin for publishing an open katana session.
+
+    This hook relies on functionality found in the base file publisher hook in
+    the publish2 app and should inherit from it in the configuration. The hook
+    setting for this plugin should look something like this::
+
+        hook: "{self}/publish_file.py:{engine}/tk-multi-publish2/basic/publish_session.py"
+
+    """
+    def __init__(self, *args, **kwargs):
+        super(KatanaLookFilePublishPlugin, self).__init__(*args, **kwargs)
+
+    @property
+    def name(self):
+        """The general name for this plugin (:class:`str`)."""
+        return "KatanaLookFilePublish"
+
+    @property
+    def icon(self):
+        """
+        The path to an icon on disk that is representative of this plugin
+        (:class:`str`).
+        """
+        return os.path.join(
+            self.disk_location,
+            "icon",
+            "look_file.png"
+        )
+    @property
+    def description(self):
+        """
+        Verbose, multi-line description of what the plugin does. This can
+        contain simple html for formatting.
+        """
+
+        return """
```

Comment:
Does this need de-indenting?
```suggestion
        from textwrap import dedent
        return dedent("""
        Publish the Look File by picking the unpublished version
        from the drop-down.

        If you can't see the version you are expecting make sure it
        hasn't already been published, or if it has been written out.
        """)
```

Reply:
Randomly, in the base class, shotgun haven't done this, so I followed their template